File: MaximumAscendingSubArraySum/Leetcode1800.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var Leetcode1800 = function (nums) {
  let prev = nums[0];
  let maxSum = nums[0];
  let currSum = nums[0];

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] > prev) {
      currSum += nums[i];
    } else {
      currSum = nums[i];
    }
    if (currSum > maxSum) {
      maxSum = currSum;
    }
    prev = nums[i];
  }
  return maxSum;
};

File: MaximumAscendingSubArraySum/Leetcode1800.cpp
#include<vector>
using std::vector;

class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int prev = nums.at(0);
        int maxSum = nums.at(0);
        int currSum = nums.at(0);
        
        for(vector<int>::iterator i = nums.begin() + 1; i != nums.end(); i++) {
            if(*i > prev) currSum += *i;
            else currSum = *i;
            if (currSum > maxSum) maxSum = currSum;
            prev = *i;
        }

        return maxSum;
    }
};